### DISCUSSION TITLE
Roadmap document

### DISCUSSION THREAD

**Discussion by yusuphgammut** [helix-editor/helix#5877] [General] [answered]
Hi folks. I've been looking into Helix in order to start using it and if possible collaborating in the project with little things for now. I think is an awesome implementation with a bright future.

However, I miss a roadmap document that outlines the main development efforts in progress and how they fit into the long term goals. I've been checking some of the open PRs and I see there's work on support for softwrap / virtual text was merged just last week. I've also seen that there's work on supporting multiple language servers per language. And I've also seen that there's the idea to add a plugin system, but I don't know if it's in progress or just an idea.

Is there a document outlining the roadmap for this project? If it isn't, shouldn't we create one and add it to the wiki? I can help with that if necessary.

**Comment by pascalkuthe** [comment, marked as answer]
There isn't really a roadmap because we are all volunteers working in our freetime on what we enjoy so there are no formal commitments really. Adding open PRs to a roadmap is sortof pointless as the existing of the PR itself is roadmap enough.

We are internally experimenting with a github board to track the progress of vsrious PRs better (bith fir ourselfs and to give people more of an idea what a pr is waiting on) in #5263 which is probably the closest we have to a roadmap

**Reply by yusuphgammut** [reply]
Thanks for the quick response. That's fair enough for me. The Github board seems like a good solution because it gives an overview of what is currently in progress, instead of checking more that 200 PRs opened. Keep the great work!

**Comment by askreet** [comment]
I'd actually like to second this request. I've been lurking around the issues and discussions for a couple weeks and even put up my first PR, but it's hard to tell what requests and such the core maintainers are interested in implementing. Some level of effort toward a roadmap doc to get an understanding of what we'd _like_ to land could be really useful to galvanizing people already willing to contribute. The document can start with a preamble explaining what @pascalkuthe states above: that this is a volunteer effort project and the items in the roadmap are not promises but rather ideals. The doc itself could link back to discussion tickets, statuses of implementations, etc.

Or perhaps something like GitHub Projects, Issues, etc. would be a better way to frame it, but still there's a lack of (or I haven't found -- apologies if that's the case) a "stuff we'd like to add" and "stuff we've ruled out" centralized anywhere.
